plugins/actuators/vllm_performance/ado_actuators/vllm_performance/deployment_management.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) IBM Corporation
+# SPDX-License-Identifier: MIT
+
+import asyncio
+import logging
+
+import ray
+
+from orchestrator.modules.operators.console_output import RichConsoleSpinnerMessage
+
+logger = logging.getLogger(__name__)
+
+
+class DeploymentWaiter:
+    def __init__(self, identifier: str):
+        self.identifier = identifier
+        self.wait_event = asyncio.Event()
+
+
+class DeploymentConflictManager:
+    def __init__(self):
+        self.deployments_to_wait_for: dict[str, DeploymentWaiter] = {}
+        self.model_already_downloaded: set[str] = set()
+
+    def maybe_add_deployment(self, model: str, identifier: str) -> bool:
+        if (
+            model not in self.model_already_downloaded
+            and model not in self.deployments_to_wait_for
+        ):
+            self.deployments_to_wait_for[model] = DeploymentWaiter(
+                identifier=identifier
+            )
+            return True
+        return False
```

Comment:
By inverting the if you can reduce nesting and improve readability:

```suggestion
        if (
            model in self.model_already_downloaded
            or model in self.deployments_to_wait_for
        ):
            return False

        self.deployments_to_wait_for[model] = DeploymentWaiter(identifier=identifier)
        return True
```